src/view.c

```diff
@@ -752,7 +755,12 @@ view_adjust_for_layout_change(struct view *view)
 		}
 	} else if (!view_apply_special_geometry(view)) {
 		/* reposition view if it's offscreen */
-		struct wlr_box box = { view->x, view->y, view->w, view->h };
+		struct wlr_box box = {
+			.x = view->pending_move_resize.x,
+			.y = view->pending_move_resize.y,
+			.width = view->pending_move_resize.width,
+			.height = view->pending_move_resize.height
+		};
```

Comment:
This could just be a `struct wlr_box box = view->geo.pending` so another point for the refactor

Reply:
Yes indeed :)

If you prefer, I can submit that pull request (for the rename/refactor) and rebase this one on top.

Reply:
Totally up to you :)

I'll put this PR into draft mode for now, but feel free to undraft it at any time.